agent/src/index.js

```diff
@@ -90,14 +77,29 @@ function isNewDocId(docId) {
   return false
 }
 
-function isNew3id(docId) {
-  // TODO: Check valid docId format
-  if (threeIdSet[docId] === undefined) {
-    threeIdSet[docId] = 1
-    return true
+function logDocId(docId) {
+  writeStream(docId, docIdLogPath, 'New docId:')
+}
+
+async function logIf3id(docId, ipfs) {
+  const payload = await getDocPayload(docId, ipfs)
+  
+  let is3id = false
+  try {
+    is3id = payload.header.tags.includes('3id')
+  } catch (error) {
+    // pass
   }
-  threeIdSet[docId]++
-  return false
+  if (is3id) {
+    writeStream(docId, threeIdLogPath, 'New 3id:')
+  }
+}
```

Comment:
Can't we just log all `tags` and `schema` here instead of only logging if the 3ID is present?
If we do we should log the values as well.
Grafana could just looks for `3id` tags to count number of 3IDs.

Reply:
resolving for now and will come back to the other header data per convo below
> @oed 3⃣ re your comment. the only thing is that im not yet sure how to label tags bc they will be arbitrary im guessing. i've been using regex for the labeling which then allows grafana to look for a specific label but i need to figure out how to create labels dynamically.
> @val 3⃣ ok, didn't know you where using a specific label feature. Then we can keep it as is for now :thumbsup: